### DISCUSSION TITLE
Run  D435i with RGB module disconnected

### DISCUSSION THREAD

**Discussion by ciniminis1** [realsenseai/realsense-ros#2382] [Q&A]
Hello there,
is there a way to run `realsense2` package with D435i sensor depth-only (with disconnected RGB module). Apparently, when the RGB module is disconnected, a following error occures: 

```
[ INFO] [1655477845.563539124]: Device with physical ID 2-4.4.1-8 was found.
[ INFO] [1655477845.563567956]: Device with name Intel RealSense D430I was found.
[ INFO] [1655477845.564035531]: Device with port number 2-4.4.1 was found.
[ INFO] [1655477845.564061695]: Device USB type: 3.2
[FATAL] [1655477845.564097679]: Unsupported device! Product ID: 0x0B4B

```

Thank you.

**Comment by xiaomengfang** [comment]
Have you solved this problem? I meter the same case when I tried to run the `roslaunch realsense2_camera rs_camera.launch` node. There is a D430i camera connected to my Jetson Xavier NX board equipped with Ubuntu 18.04.
<img width="712" alt="image" src="https://user-images.githubusercontent.com/68259399/229026959-d4722024-b948-4174-8c46-6b0cba3e8a79.png">

**Comment by MartyG-RealSense** [comment]
Hi @fangxiaomeng0830  A RealSense camera's firmware driver identifies the name of the camera model based on the hardware components that it can detect.  When a D435i's RGB sensor is not detected by the firmware it is assigned the name D430i, because the camera contains a D430 camera circuit board and an IMU component.

The D430i state is typically caused by a problem with the firmware driver that can be corrected by downgrading the firmware to the previous version of the driver that is currently installed and then re-upgrading to your current one.

**Reply by xiaomengfang** [reply]
Thank you for your kind response！I'm new to real sense camera, could you please tell me how to alter the firmware? By the way, the camera is a module integrated to a quadruped robot (Unitree-B1), I'm not sure it is D435i or not and just incorrectly detected as D430i.

![image](https://user-images.githubusercontent.com/68259399/229038917-223c4ce1-fdcd-4afc-abac-e51914ab2b1e.png)

**Comment by MartyG-RealSense** [comment]
If the RealSense camera is an integrated one then the manufacturer of the product that contains the camera should be contacted for support as Intel cannot provide support for such cameras unfortunately.  

The official store website of Unitree B1 says that it is equipped with **D430x5**.  D430 is not equipped with an RGB sensor, so it appears that the absence of the RGB sensor may be deliberate and not a firmware fault.

https://shop.unitree.com/en-gb/products/unitree-b1

**Reply by xiaomengfang** [reply]
Thanks, @MartyG-RealSense. I wonder that if there are ROS wrappers that support D430i? Or can I only do development based on the official SDK?

**Comment by MartyG-RealSense** [comment]
In the absence of an RGB sensor, a D430 can use an infrared image instead of an RGB image as a pointcloud texture by adding the command below to the end of the ROS launch instruction.

`pointcloud_texture_stream:=RS2_STREAM_ANY`

However, I understand your problem that the D430i identification is causing a red error when launching.  I will ask the RealSense ROS development team whether it is possible to add D430i as a supported PID for situations such as yours.

**Reply by xiaomengfang** [reply]
Thanks for your help! @MartyG-RealSense.  I'm looking forward to your reply.

**Reply by xiaomengfang** [reply]
Dear @MartyG-RealSense , I find that the D430i was connected to the NX board using a 2.1 port, and there appears a [WARN]. I want to know that if the type of the communication interface will influence the ROS wrapper.

**Reply by Nir-Az** [reply]
Merged PR #2322 , D430i PID is now being recognized on ROS2-development branch.
@SamerKhshiboun  FYI

**Comment by MartyG-RealSense** [comment]
The ROS wrapper can be used with a USB 2.1 port, though the number of supported resolution and FPS combinations will be limited compared to a USB 3 connection.  USB 2.1 is also slower and the Infrared 2 stream (right-side infrared) will not be available as it is a USB 3 feature.  For 2.1 I recommend using 640x480 resolution and 15 or 30 FPS.

**Reply by xiaomengfang** [reply]
Thanks, so what's the potential reason of the red [FATAL] problem I met, and how I could solve it? Plus, whether the Jetson Xavier Nx board will influence the ROS wrapper?

**Comment by MartyG-RealSense** [comment]
Jetson boards including Xavier NX work with the RealSense ROS wrapper.

The red error is likely simply due to the D430i model number being unsupported as it does not usually occur unless there is a fault such as a damaged firmware or a disconnected RGB sensor cable inside the camera.

**Reply by xiaomengfang** [reply]
The picture below is the log file when start the ROS Node.

<img width="722" alt="image" src="https://user-images.githubusercontent.com/68259399/229109533-3a5bd904-225d-4b8e-89ee-45590fb20a70.png">

**Comment by MartyG-RealSense** [comment]
**Bond broken** is a rare error.  One RealSense ROS user solved it by reducing resolution with this command:

roslaunch realsense2_camera rs_camera.launch color_width:=424 color_height:=240 color_fps:=15 depth_width:=424 depth_height:=240 depth_fps:=15
